webapp/components/expression/expressionVariables.js

```diff
@@ -69,24 +71,55 @@ const getChildDefVariables = ({ survey, nodeDefContext, nodeDefCurrent, mode, la
       }
     }
   }
+
   return variables
 }
 
+export const getVariablesGroupedByParentUuid = ({ variables, survey: surveyParam }) => {
+  const survey = Survey.buildAndAssocDependencyGraph(surveyParam)
+
+  const variablesGroupedByParentUuid = (variables || []).reduce(
+    (byParentUuid, variable) => ({
+      ...byParentUuid,
+      [variable.parentUuid]: [...(byParentUuid[variable.parentUuid] || []), variable],
+    }),
+    {}
+  )
+
+  const variablesGrouped = Object.keys(variablesGroupedByParentUuid)
+    .map((parentUuid) => {
+      const parentNodeDef = Survey.getNodeDefByUuid(parentUuid)(survey)
+      return { parentUuid, label: NodeDef.getName(parentNodeDef), id: Number(NodeDef.getId(parentNodeDef)) }
+    })
+    .sort((groupA, groupB) => (groupA.id > groupB.id ? 1 : -1))
```

Comment:
why are they sorted by id? should they be sorted by label or name or by hierarchy level instead? I prefer the hierarchy level

Reply:
The same order as the Form is required, if we use hierarchy this level is not going to be the same, I am not sure if it works in the future with the ID, do we have any way to get the order?

Reply:
the order in the form is given by layout prop, not IDs.

Reply:
Not sure about how to sort elements using these layout props, do we have any method to have the position?

Reply:
we dont have a method now. you could eventually implement it now, or we'll handle the sorting separately later on.
up to you, but let us know so that we can create an issue in case we'll do it later

Reply:
I added a method,
not sure if the name is good enough.
I will leave a comment there